compose/config/config.py

```diff
@@ -165,13 +164,11 @@ def find_candidates_in_parent_dirs(filenames, path):
     return (candidates, path)
 
 
-@validate_top_level_object
-@validate_service_names
 def pre_process_config(config):
-    """
-    Pre validation checks and processing of the config file to interpolate env
+    """Pre validation checks and processing of the config file to interpolate env
     vars returning a config dict ready to be tested against the schema.
     """
+    validate_top_level_object(config)
```

Comment:
Ya, i was going back and forth on this one. This is also called from the `extends` resolving.  I was considering just dropping this method and calling both functions. I think I'll do that.